lib/core/device.rb

```diff
@@ -707,6 +707,114 @@ def swipe_down(action)
       .perform
   end
 
+  # Scrolls vertically until the specified element is visible or a timeout is reached
+  # Accepts:
+  #   Strategy
+  #   Id
+  #   ScrollTarget
+  #     Strategy
+  #     Id
+  #     RecheckAfterScrolls
+  #   SwipeAction
+  #     OffsetFractionX
+  #     StartFractionY
+  #     EndFractionY
+  #     SwipeSpeedMultiplier
+  #     SwipePauseDuration
+  #   ScrollTimeout
+  #   FullView
+  #   FullViewOffsetY
+  def scroll_until_element_visible(action)
+    # the element will be checked in a loop, so its search should be fast and never fail
+    original_noraise = action["NoRaise"]
+    action["NoRaise"] = true
+    action["Time"] = 0.5
+    # configure swipe action properties
+    x_frac = 0.5
+    y_start_frac = 0.7
+    y_end_frac = 0.3
+    scroll_mul = @platform == "iOS" ? 3 : 1.5
+    scroll_pause = @platform == "iOS" ? 0.2 : 0.1
```

Comment:
whats the reason for using double on iOS?

Reply:
This way the behavior for both OS is visually similar. For the multiplier, I think this may be related to Appium's retrieved dimensions actually being half of the real dimensions (pixel-wise), so the scrolling speed must also be changed. For pause, it was probably due to different minimum thresholds to not cause overscroll.